yarn-project/acir-simulator/src/client/client_execution_context.ts

```diff
@@ -436,4 +436,29 @@ export class ClientExecutionContext extends ViewDataOracle {
       startSideEffectCounter,
     );
   }
+
+  /**
+   * Read the public storage data.
+   * @param startStorageSlot - The starting storage slot.
+   * @param numberOfElements - Number of elements to read from the starting storage slot.
+   */
+  public async storageRead(startStorageSlot: Fr, numberOfElements: number): Promise<Fr[]> {
+    // TODO(#4320): This is a hack to work around not having directly access to the public data tree but
+    // still having access to the witnesses
```

Comment:
Why not just expose the tree directly? Is it simply to get something to work before we decide to invest more time in this?

Reply:
Look at #4320, it is because there need to be more plumbing in place. We straight up don't have access to the data through the node right now (in private execution).